Libraries/LibURL/Parser.cpp

```diff
@@ -881,7 +881,18 @@ URL Parser::basic_parse(StringView raw_input, Optional<URL> const& base_url, URL
     bool inside_brackets = false;
     bool password_token_seen = false;
 
-    Utf8View input(processed_input);
+    auto scalar_value_input = ({
+        // To convert a string into a scalar value string, replace any surrogates with U+FFFD.
```

Comment:
It would be nice if we had a common function for this between here and convert_to_scalar_string from Infra/Strings.h

e.g, an optimization which was applied to some other functions was to have a fast-path if the utf8_view was already a valid string (calling `validate()`), and I think the same would likely apply here since it's going to be valid unicode 99% of the time.

Also, could we please have a test case?

But of course this also goes away with a USVSString wrapper as well.

Reply:
My proposal is to split `Utf8View` (and a few other `String` classes) into a validated and unvalidated form.